lib/optimizely/decision_service.rb

```diff
@@ -121,16 +112,14 @@ def get_variation_for_feature(feature_flag, user_id, attributes = nil)
 
       # check if the feature is being experiment on and whether the user is bucketed into the experiment
       decision = get_variation_for_feature_experiment(feature_flag, user_id, attributes)
-      unless decision.nil?
-        return decision
-      end
+      return decision unless decision.nil?
 
       feature_flag_key = feature_flag['key']
       variation = get_variation_for_feature_rollout(feature_flag, user_id, attributes)
       if variation
         @config.logger.log(
           Logger::INFO,
-          "User '#{user_id}' is in the rollout for feature flag '#{feature_flag_key}'."
+          "User '#{user_id}' is bucketed into a rollout for feature flag '#{feature_flag_key}'."
```

Comment:
describe the actual rule the user was bucketed into by logging the audience name as the rule name.

Reply:
You probably meant to have audience name logged inside the get variation for feature rollout and not at this place. I have put it where you have done in the java sdk